sklearn/cluster/tests/test_spectral.py

```diff
@@ -205,3 +205,163 @@ def test_spectral_clustering_with_arpack_amg_solvers():
         assert_raises(
             ValueError, spectral_clustering,
             graph, n_clusters=2, eigen_solver='amg', random_state=0)
+
+
+def test_6489_regression():
+    X=np.array([[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
```

Comment:
Spaces around = please.

Could you use 

```py
X = np.ones((50, 50))
X[[ 3,  3,  3,  3,  3,  3,  8,  8,  8,  8,  8,  8, 17, 17, 17, 17, 17,
   17, 33, 33, 33, 33, 33, 33, 34, 34, 34, 34, 34, 34, 35, 35, 35, 35,
   35, 35, 45, 45, 45, 45, 45, 45, 46, 46, 46, 46, 46, 46],
  [17, 33, 34, 35, 45, 46, 17, 33, 34, 35, 45, 46,  3,  8, 33, 34, 35,
   46,  3,  8, 17, 35, 45, 46,  3,  8, 17, 35, 45, 46,  3,  8, 17, 33,
   34, 45,  3,  8, 33, 34, 35, 46,  3,  8, 17, 33, 34, 45]] = 0
```

for brevity?

Reply:
yes !! ( I just copy pasted what was in the issue )